remote_execution/oss/re_grpc/src/client.rs

```diff
@@ -414,11 +419,42 @@ pub struct GRPCClients {
     capabilities_client: CapabilitiesClient<InterceptedService<Channel, InjectHeadersInterceptor>>,
 }
 
+enum DigestRemoteState {
+    ExistsOnRemote,
+    Missing
+}
+
+struct FindMissingCache {
+    cache: LruCache<TDigest, DigestRemoteState>,
+    ttl: Duration,
+    last_check: Instant,
```

Comment:
```suggestion
    /// To avoid a situation where we cache that an artifact is available remotely, but the artifact then expires
    /// we clear our local cache once every `ttl`.
    ttl: Duration,
    last_clear: Instant,
```